src/box/lua/config/applier/autoexpel.lua

```diff
@@ -0,0 +1,214 @@
+local log = require('internal.config.utils.log')
+
+-- {{{ General-purpose utils
+
+-- {'a', 'b', 'c'} => {a = true, b = true, c = true}
+local function array2set(t)
+    local res = {}
+    for _, v in ipairs(t) do
+        res[v] = true
+    end
+    return res
+end
+
+-- }}} General-purpose utils
+
+-- {{{ box status notifier
+
+local _box_status_watcher
+
+-- Invoke the given callback when box.status event occurs.
+--
+-- It reacts on RO->RW, RW->RO switch and the data dictionary
+-- schema version update.
+local function box_status_notifier_start(cb)
+    if _box_status_watcher ~= nil then
+        return
+    end
+
+    _box_status_watcher = box.watch('box.status', function(_status)
+        cb()
+    end)
+end
+
+-- Stop to call the callback and free resources.
+local function box_status_notifier_stop()
+    if _box_status_watcher == nil then
+        return
+    end
+
+    _box_status_watcher:unregister()
+    _box_status_watcher = nil
+end
+
+-- }}} box status notifier
+
+-- {{{ Hold last config
+
+local _last_cfg
```

Comment:
nit: I prefer all the applier state variables to be stored in one table, this makes it easier to expand the logic

Reply:
It has cons and pros. I like that now I have such a mini-modules, where all the coupled things (state and functions) are placed together.

I had an idea to make it more explicit by storing the state in a table and transform the functions into methods, but then decided that I don't need it here and shouldn't make things unnecessarily complicated.

So, I would keep it as is.

Reply:
sure, your approach makes sense too!